### PR TITLE
Run test pods as a non-privileged users

### PR DESCRIPTION
**Pull request by thilinapiy**
With this PR users will be able to run helm chart test hook containers as a non-privileged user.

- Use `alpine/curl` image instead of `alpine` image and no need to install curl package at the runtime.
- Inherit security context from main app.
- Inherit `imagePullSecrets` from main app.
- Disable test if the main app is disabled.

### CODE REVIEW

**Comment by mtyazici** [comment]
Hi @thilinapiy, could you bump the Chart version for both charts?

**Comment by thilinapiy** [comment]
Hi @mtyazici,
I have update the Chart versions for both PRs.
Cheers

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml [hunk 1]
```diff
@@ -13,22 +14,40 @@ metadata:
...
-  restartPolicy: Never
\ No newline at end of file
+    resources:
+{{ toYaml .Values.test.resources | indent 6 }}
+    {{- if .Values.test.securityContext.enabled }}
+    securityContext:
+      runAsNonRoot: {{ if eq (int .Values.test.securityContext.runAsUser) 0 }}false{{ else }}true{{ end }}
+      runAsUser: {{ .Values.test.securityContext.runAsUser }}
+      runAsGroup: {{ .Values.test.securityContext.runAsGroup }}
+      privileged: false
```
Comment (hakanmemisoglu) [inline stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml:50]: Where is `privileged` coming from? 
I could not see it in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#podsecuritycontext-v1-core. 
If it is Openshift related, it should be templatized.

Comment (hakanmemisoglu) [inline stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml:50]: Where is `privileged` coming from? 
I could not see it in https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#podsecuritycontext-v1-core. 
If it is Openshift related, it should be templatized.

File: stable/hazelcast-enterprise/templates/tests/test-management-center.yaml [hunk 2]
```diff
@@ -35,4 +44,14 @@ spec:
...
-  restartPolicy: Never
\ No newline at end of file
+    resources:
+{{ toYaml .Values.test.resources | indent 6 }}
+    {{- if .Values.test.securityContext.enabled }}
+    securityContext:
+      runAsNonRoot: {{ if eq (int .Values.test.securityContext.runAsUser) 0 }}false{{ else }}true{{ end }}
+      runAsUser: {{ .Values.test.securityContext.runAsUser }}
+      runAsGroup: {{ .Values.test.securityContext.runAsGroup }}
+      privileged: false
```
Comment (hakanmemisoglu) [inline stable/hazelcast-enterprise/templates/tests/test-management-center.yaml:54]: Same here.

Comment (hakanmemisoglu) [inline stable/hazelcast-enterprise/templates/tests/test-management-center.yaml:54]: Same here.

File: stable/hazelcast/templates/tests/test-hazelcast.yaml [hunk 3]
```diff
@@ -13,22 +14,40 @@ metadata:
...
-  restartPolicy: Never
\ No newline at end of file
+    resources:
+{{ toYaml .Values.test.resources | indent 6 }}
+    {{- if .Values.test.securityContext.enabled }}
+    securityContext:
+      runAsNonRoot: {{ if eq (int .Values.test.securityContext.runAsUser) 0 }}false{{ else }}true{{ end }}
+      runAsUser: {{ .Values.test.securityContext.runAsUser }}
+      runAsGroup: {{ .Values.test.securityContext.runAsGroup }}
+      privileged: false
```
Comment (hakanmemisoglu) [inline stable/hazelcast/templates/tests/test-hazelcast.yaml:50]: Same here.

Comment (hakanmemisoglu) [inline stable/hazelcast/templates/tests/test-hazelcast.yaml:50]: Same here.

File: stable/hazelcast/templates/tests/test-management-center.yaml [hunk 4]
```diff
@@ -35,4 +44,14 @@ spec:
...
-  restartPolicy: Never
\ No newline at end of file
+    resources:
+{{ toYaml .Values.test.resources | indent 6 }}
+    {{- if .Values.test.securityContext.enabled }}
+    securityContext:
+      runAsNonRoot: {{ if eq (int .Values.test.securityContext.runAsUser) 0 }}false{{ else }}true{{ end }}
+      runAsUser: {{ .Values.test.securityContext.runAsUser }}
+      runAsGroup: {{ .Values.test.securityContext.runAsGroup }}
+      privileged: false
```
Comment (hakanmemisoglu) [inline stable/hazelcast/templates/tests/test-management-center.yaml:54]: Same here.

Comment (hakanmemisoglu) [inline stable/hazelcast/templates/tests/test-management-center.yaml:54]: Same here.

**Comment by hakanmemisoglu** [comment]
@thilinapiy I dropped some comments. Can you also resolve the merge conflicts?

**Comment by hakanmemisoglu** [comment]
@thilinapiy I dropped some comments. Can you also resolve the merge conflicts?

File: stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml (hunk 1 above)
Comment (thilinapiy) [reply stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml:50]: It is coming from `SecurityContext`. aAs I know this get more precedence over `PodSecurityContext`.
https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#securitycontext-v1-core

BTW I just copied it from an existing template: https://github.com/hazelcast/charts/blob/master/stable/hazelcast-enterprise/templates/mancenter-statefulset.yaml#L230

IMO this should be fine. We can parameterise that if needed. What do you think?

Comment (hakanmemisoglu) [reply stable/hazelcast-enterprise/templates/tests/test-hazelcast.yaml:50]: OK, got it. Thanks for the explanation. Let me check it one more time.

**Review by mtyazici — APPROVED** [review]

**Review by hakanmemisoglu — APPROVED** [review]

**Pull request merged**